rust/zephyr/src/gpio.rs

```diff
@@ -0,0 +1,950 @@
+            3 => Some(Self::P3),
+            4 => Some(Self::P4),
+            5 => Some(Self::P5),
+            6 => Some(Self::P6),
+            7 => Some(Self::P7),
+            8 => Some(Self::P8),
+            9 => Some(Self::P9),
+            10 => Some(Self::P10),
+            11 => Some(Self::P11),
+            12 => Some(Self::P12),
+            13 => Some(Self::P13),
+            14 => Some(Self::P14),
+            15 => Some(Self::P15),
+            16 => Some(Self::P16),
+            17 => Some(Self::P17),
+            18 => Some(Self::P18),
+            19 => Some(Self::P19),
+            20 => Some(Self::P20),
+            21 => Some(Self::P21),
+            22 => Some(Self::P22),
+            23 => Some(Self::P23),
+            24 => Some(Self::P24),
+            25 => Some(Self::P25),
+            26 => Some(Self::P26),
+            27 => Some(Self::P27),
+            28 => Some(Self::P28),
+            29 => Some(Self::P29),
+            30 => Some(Self::P30),
+            31 => Some(Self::P31),
+            _ => None,
+        }
+    }
+}
+
+bitflags! {
+    pub struct Pins: u32 {
+        const P0  = (1_u32 << Pin::P0 as u32);
+        const P1  = (1_u32 << Pin::P1 as u32);
+        const P2  = (1_u32 << Pin::P2 as u32);
+        const P3  = (1_u32 << Pin::P3 as u32);
+        const P4  = (1_u32 << Pin::P4 as u32);
+        const P5  = (1_u32 << Pin::P5 as u32);
+        const P6  = (1_u32 << Pin::P6 as u32);
+        const P7  = (1_u32 << Pin::P7 as u32);
+        const P8  = (1_u32 << Pin::P8 as u32);
+        const P9  = (1_u32 << Pin::P9 as u32);
+        const P10 = (1_u32 << Pin::P10 as u32);
+        const P11 = (1_u32 << Pin::P11 as u32);
+        const P12 = (1_u32 << Pin::P12 as u32);
+        const P13 = (1_u32 << Pin::P13 as u32);
+        const P14 = (1_u32 << Pin::P14 as u32);
+        const P15 = (1_u32 << Pin::P15 as u32);
+        const P16 = (1_u32 << Pin::P16 as u32);
+        const P17 = (1_u32 << Pin::P17 as u32);
+        const P18 = (1_u32 << Pin::P18 as u32);
+        const P19 = (1_u32 << Pin::P19 as u32);
+        const P20 = (1_u32 << Pin::P20 as u32);
+        const P21 = (1_u32 << Pin::P21 as u32);
+        const P22 = (1_u32 << Pin::P22 as u32);
+        const P23 = (1_u32 << Pin::P23 as u32);
+        const P24 = (1_u32 << Pin::P24 as u32);
+        const P25 = (1_u32 << Pin::P25 as u32);
+        const P26 = (1_u32 << Pin::P26 as u32);
+        const P27 = (1_u32 << Pin::P27 as u32);
+        const P28 = (1_u32 << Pin::P28 as u32);
+        const P29 = (1_u32 << Pin::P29 as u32);
+        const P30 = (1_u32 << Pin::P30 as u32);
+        const P31 = (1_u32 << Pin::P31 as u32);
+
+        const PINS_0_TO_31  = 0xFFFFFFFF;
+        const PINS_0_TO_15  = 0x0000FFFF;
+        const PINS_15_TO_31 = 0xFFFF0000;
+        const PINS_0_TO_7   = 0x000000FF;
+        const PINS_8_TO_15  = 0x0000FF00;
+        const PINS_16_TO_23 = 0x00FF0000;
+        const PINS_24_TO_31 = 0xFF000000;
```

Comment:
Do you expect clients will use these group definitions? Maybe we should keep the initial API minimal.